frontends/uhdm/UhdmAst.cc

```diff
@@ -85,16 +85,29 @@ void UhdmAst::visit_default_expr(vpiHandle obj_h)  {
 		auto assign_node = new AST::AstNode(AST::AST_ASSIGN_EQ);
 		auto id_node = new AST::AstNode(AST::AST_IDENTIFIER);
 		id_node->str = current_node->str;
+
+		auto always_child = mod->children.begin();
 		for (auto child : mod->children) {
 			if (child->type == AST::AST_INITIAL) {
 				initial_node = child;
+			}
+
+			if (initial_node != nullptr && (*always_child)->type == AST::AST_ALWAYS)
 				break;
+
+			if ((*always_child)->type != AST::AST_ALWAYS) {
+				always_child++;
 			}
```

Comment:
```suggestion
		AST::AstNode* always_child = nullptr;
		for (auto child : mod->children) {
			if (child->type == AST::AST_INITIAL) {
				initial_node = child;
				break;
			} else if (child->type == AST::AST_ALWAYS) {
			        always_child = child;
			        break;
			}
```

We want to add initial node before any another always node, so if we found, that there is already initial node in module, we should stop looking for always and only when we won't find initial node, we need location of always node.

Reply:
I modified the code to stop looking for AST_ALWAYS after it has found AST_INITIAL. I did it in a different way because `always_child` must be an iterator that can be used in `insert()`

Reply:
Ok, so I think it would be better to use ``find_if`` in ``if`` below, when there isn't AST_INITIAL node to get the iterator.
Current code is very hard to follow (at least to me) and even variable is named ``always_child`` it is not guaranteed, that behind this iterator there is ``AST_ALWAYS`` node, so it is very confusing.

Reply:
Simplified with `find_if`. Now we just look for `AST_INITIAL` in the loop. Then if it was not found we get the iterator for first `AST_ALWAYS` or for the last child of `AST_MODULE` if AST_ALWAYS was not found. Then we insert the AST_INTIAL at the place of the iterator.